src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -63,6 +64,321 @@ internal override string ExtensionName
             }
         }
 
+        /// <summary>
+        /// This name uses an IL-looking format to encode CLR-level information of an extension block (ie. arity, constraints, extended type).
+        /// It is meant be to hashed to produce the content-based name for the extension grouping type.
+        /// </summary>
+        internal string ComputeExtensionGroupingRawName()
+        {
+            var pooledBuilder = PooledStringBuilder.GetInstance();
+            var builder = pooledBuilder.Builder;
+            builder.Append("extension");
+
+            if (this.IsGenericType)
+            {
+                builder.Append("<");
+
+                foreach (var typeParameter in this.TypeParameters)
+                {
+                    if (typeParameter.Ordinal > 0)
+                    {
+                        builder.Append(", ");
+                    }
+
+                    appendTypeParameter(typeParameter, builder, isDefinition: true);
+                }
+
+                builder.Append(">");
+            }
+
+            builder.Append("(");
+            if (this.ExtensionParameter is { } extensionParameter)
+            {
+                appendType(extensionParameter.Type, builder);
+            }
+
+            builder.Append(")");
+
+            return pooledBuilder.ToStringAndFree();
+
+            static void appendType(TypeSymbol type, StringBuilder builder)
+            {
+                if (type is NamedTypeSymbol namedType)
+                {
+                    appendNamedType(type, builder, namedType);
+                }
+                else if (type is TypeParameterSymbol typeParameter)
+                {
+                    appendTypeParameter(typeParameter, builder, isDefinition: false);
```

Comment:
>typeParameter [](http://example.com/codeflow?start=40&length=13)

In an error scenario this could be a type parameter of an enclosing type. Right?